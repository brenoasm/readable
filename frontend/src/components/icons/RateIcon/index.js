import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '../Icon';

const propTypes = {
  onClick: PropTypes.func,
  title: PropTypes.string,
  rateDown: PropTypes.bool
};

const defaultProps = {
  onClick: () => {},
  title: '',
  rateDown: false,
};

const StyledRateIcon = styled.span`
  font-size: x-large;
  cursor: pointer;
`;

const RateIcon = ({
  onClick,
  rateDown,
  title
}) => (
  <StyledRateIcon>
    <Icon
      title={title}
      onClick={onClick}
      icon={rateDown ? "fas fa-caret-down" : "fas fa-caret-up" }
    />
  </StyledRateIcon>
);

RateIcon.propTypes = propTypes;
RateIcon.defaultProps = defaultProps;

export default RateIcon;
