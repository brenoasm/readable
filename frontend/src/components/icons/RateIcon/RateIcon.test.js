/* eslint-disable */

import React from 'react';

import RateIcon from './';

describe('RateIcon', () => {
  it('matches the snapshot (rate up)', () => {
    const initialProps = {
      onClick: jest.fn(),
      title: 'Thumbs up!'
    };

    const wrapper = shallow(<RateIcon {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot (rate down)', () => {
    const initialProps = {
      onClick: jest.fn(),
      rateDown: true,
      title: 'Thumbs down!'
    };

    const wrapper = shallow(<RateIcon {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
