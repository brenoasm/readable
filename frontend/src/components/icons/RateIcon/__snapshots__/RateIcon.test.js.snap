// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateIcon matches the snapshot (rate down) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RateIcon
    onClick={[Function]}
    rateDown={true}
    title="Thumbs down!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        icon="fas fa-caret-down"
        onClick={[Function]}
        title="Thumbs down!"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "fas fa-caret-down",
        "onClick": [Function],
        "title": "Thumbs down!",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          icon="fas fa-caret-down"
          onClick={[Function]}
          title="Thumbs down!"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "fas fa-caret-down",
          "onClick": [Function],
          "title": "Thumbs down!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RateIcon matches the snapshot (rate up) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RateIcon
    onClick={[Function]}
    rateDown={false}
    title="Thumbs up!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        icon="fas fa-caret-up"
        onClick={[Function]}
        title="Thumbs up!"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "fas fa-caret-up",
        "onClick": [Function],
        "title": "Thumbs up!",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          icon="fas fa-caret-up"
          onClick={[Function]}
          title="Thumbs up!"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "fas fa-caret-up",
          "onClick": [Function],
          "title": "Thumbs up!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
